EasyPost.Tests/ServicesTests/Beta/RateServiceTest.cs

```diff
@@ -0,0 +1,85 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using EasyPost.Calculation;
+using EasyPost.Models.API.Beta;
+using EasyPost.Tests._Utilities;
+using EasyPost.Tests._Utilities.Annotations;
+using EasyPost.Utilities.Annotations;
+using Xunit;
+
+namespace EasyPost.Tests.ServicesTests.Beta
+{
+    public class RateServiceTests : UnitTest
+    {
+        public RateServiceTests() : base("beta_rate_service")
+        {
+        }
+
+        #region Tests
+
+        #region Test CRUD Operations
+
+        [Fact]
+        [CrudOperations.Read]
+        [Testing.Function]
+        public async Task TestRetrieve()
+        {
+            UseVCR("retrieve");
+
+            Dictionary<string, object> data = Fixtures.BasicShipment;
+
+            List<StatelessRate> rates = await Client.Beta.Rate.RetrieveStatelessRates(data);
+
+            Assert.IsType<List<StatelessRate>>(rates);
+        }
+
+        #endregion
+
+        [Fact]
+        [Testing.Function]
+        public async Task TestGetLowestRateInstanceFunction()
+        {
+            UseVCR("get_lowest_rate_instance_function");
+
+            Dictionary<string, object> data = Fixtures.BasicShipment;
+
+            List<StatelessRate> rates = await Client.Beta.Rate.RetrieveStatelessRates(data);
+
+            StatelessRate lowestStatelessRate = Calculation.Rates.GetLowestStatelessRate(rates);
+
+            Assert.Equal("First", lowestStatelessRate.Service);
+        }
+
+        [Fact]
+        [Testing.Function]
+        public async Task TestGetLowestRateStaticFunction()
+        {
+            UseVCR("get_lowest_rate_static_function");
+
+            Dictionary<string, object> data = Fixtures.BasicShipment;
+
+            List<StatelessRate> rates = await Client.Beta.Rate.RetrieveStatelessRates(data);
+
+            StatelessRate lowestStatelessRate = Client.Beta.Rate.GetLowestStatelessRate(rates);
```

Comment:
Let's have this only available as a util, like in Python since the client isn't needed for this standalone helper function

Reply:
Should I remove the non-API filtering functions from the normal Rate service (filtering rates and smart rates) as well?

Reply:
No, removing things is a breaking change

Reply:
https://github.com/EasyPost/easypost-csharp/pull/398#discussion_r1088365090